src/main/webapp/3Dscript.js

```diff
@@ -152,38 +189,58 @@ function plotOrientation() {
         THREE.Math.degToRad(point.rollDeg)));
     matrix.multiply(new THREE.Matrix4().makeTranslation(0.0, .05, 0.0));
-    orientation.setMatrixAt(i, matrix);
+    orientation.setMatrixAt(pose.indexOf(point), matrix);
   }
   orientation.instanceMatrix.needsUpdate = true;
-  orientation.position.x = poseTransform.translateX;
-  orientation.position.z = poseTransform.translateZ;
-  orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
+  orientation.position.x = transforms[runId].translateX;
+  orientation.position.z = transforms[runId].translateZ;
+  orientation.rotation.y = THREE.Math.degToRad(transforms[runId].rotate);
 }
 
 /**
- * This function initializes gui allowing the user to manipulate the
- * pose objects.
+ * This function initializes gui allowing the user to manipulate the pose
+ * objects.
  */
-function gui() {
+function loadGui() {
+  // Need a way to select which pose run to make this change to and pass this
+  // parameter.
+  const runId = currentId.value;
   const gui = new GUI();
-  gui.add(poseTransform, 'rotate', 0, 360, 1).onChange(plotOrientation)
-      .onFinishChange(plotTrajectory)
+  console.log(runs);
+  let currentData = runs[runId].data;
+  // On change of selected runId, update the given dataset to use;
+  gui.add(currentId, 'value', Object.keys(runs)).onFinishChange(
+      () => {
+        currentData = runs[currentId.value].data; console.log(currentId);
+      });
+
+  gui.add(transforms[currentId.value], 'rotate', 0, 360, 1)
+      .onChange(() => {
+        console.log('changed'); plotOrientation(currentId.value);
+      })
+      .onFinishChange(() => {
+        console.log(currentId.value);
+        lotTrajectory(currentId.value, currentData);
+      })
       .name('Pose Rotation (degrees)');
-  gui.add(poseTransform, 'translateX', -10, 10, .025)
-      .onChange(plotOrientation)
-      .onFinishChange(plotTrajectory).name('X Axis Translation');
-  gui.add(poseTransform, 'translateZ', -10, 10, .025)
-      .onChange(plotOrientation)
-      .onFinishChange(plotTrajectory).name('Z Axis Translation');
-  gui.add(poseTransform, 'scale', .5, 2, .25)
-      .onChange(plotOrientation)
-      .onFinishChange(plotTrajectory).name('Pose Scale Multiplier');
+  gui.add(transforms[currentId.value], 'translateX', -10, 10, .025)
+      .onChange(() => plotOrientation(currentId.value))
+      .onFinishChange(() => plotTrajectory(currentId.value, currentData))
+      .name('X Axis Translation');
+  gui.add(transforms[currentId.value], 'translateZ', -10, 10, .025)
+      .onChange(() => plotOrientation(currentId.value))
+      .onFinishChange(() => plotTrajectory(currentId.value, currentData))
+      .name('Z Axis Translation');
+  gui.add(transforms[currentId.value], 'scale', .5, 2, .25)
+      .onChange(() => plotOrientation(currentId.value))
+      .onFinishChange(() => plotTrajectory(currentId.value, currentData))
+      .name('Pose Scale Multiplier');
```

Comment:
question us three can discuss later, but should we have multiple guis? rn its alot going on it. and we dont want it to be too complicated.

Reply:
I agree this is getting complicated, I'm not totally sure what our approach should be or whether it is possible to create multiple dat.gui instances on one page. We certainly could try it so our display doesn't get too cluttered. Luckily this code only adds one additional text box to select the run so not too much clutter when actually rendered.